internal/cmd/rootcmd/config/cli_validation_test.go

```diff
@@ -156,3 +158,24 @@ func TestConfigValidatedVars(t *testing.T) {
 		}
 	}
 }
+
+func TesCLIArgumentParser(t *testing.T) {
+	c := config.NewCLIConfig()
+	flagSet := c.FlagSet()
+	flagSet.SetOutput(io.Discard)
+
+	err := flagSet.Parse([]string{
+		"--feature-gates=GatewayAlpha=true,RewriteURIs=true",
+		"--health-probe-bind-address=:4321",
+	})
+	require.NoError(t, err)
+	require.Equal(t, true, c.FeatureGates.Enabled(mgrconfig.GatewayAlphaFeature))
+	require.True(t, c.FeatureGates.Enabled(mgrconfig.RewriteURIsFeature))
+	for key, value := range mgrconfig.GetFeatureGatesDefaults() {
+		if key == mgrconfig.GatewayAlphaFeature || key == mgrconfig.RewriteURIsFeature {
+			continue
+		}
+		require.Equal(t, value, c.FeatureGates.Enabled(key))
+	}
+	require.Equal(t, c.ProbeAddr, ":4321")
```

Comment:
How about refactoring this into a typical table-driven test and verifying other cases like "empty feature gates", "setting a default on flag to false"?